src/ape/api/providers.py

```diff
@@ -1216,12 +1216,24 @@ def _create_call_tree_node(
 
     @classmethod
     def _create_trace_frame(cls, evm_frame: EvmTraceFrame) -> TraceFrame:
+        address_bytes = evm_frame.address
+        if address_bytes:
+            address_str = address_bytes.hex()
+            try:
+                address = cls.network.ecosystem.decode_address(address_str)
+            except Exception as err:
+                logger.warning(f"Address '{address_str}' is not checksummed (err={err}).")
+                address = cast(AddressType, address_str)
```

Comment:
am generally nervous about checksumming addresses since it fails during input data sometimes but i guess here it should always be valid so i can remove this